src/main/java/Duke.java

```diff
@@ -1,10 +1,138 @@
+        System.out.println(LINE);
+    }
+    public static void processInput(String input) {
+        String inst = input.split("\\s+")[0];
+        if (input.equalsIgnoreCase("list")) {
+            printList();
+            return;
+        } else if (inst.equalsIgnoreCase("mark")) {
+            markTask(input);
+        } else if (inst.equalsIgnoreCase("unmark")) {
+            unmarkTask(input);
+        } else if (inst.equalsIgnoreCase("todo")) {
+            todoTask(input);
+        } else if(inst.equalsIgnoreCase("deadline")) {
+            deadlineTask(input);
+        } else if(inst.equalsIgnoreCase("event")) {
+            eventTask(input);
+        } else {
+            unrecognisedInput();
+        }
+    }
+    private static void printList() {
+        if(list.size() == 0) {
+            System.out.println(LINE);
+            System.out.println("Nothing on your list yet!");
+            System.out.println(LINE);
+        } else {
+            System.out.println(LINE);
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < list.size(); i += 1) {
+                System.out.println(Integer.toString(i+1) + "." + list.get(i));
+            }
+            System.out.println(LINE);
+        }
+    }
+    private static void markTask(String input) {
+        int index = Integer.parseInt(input.split("\\s+")[1]) - 1;
+        if (indexNotInList(index)) { return; }
+        list.get(index).markDone();
+        markResponse(index);
+    }
+    private static void markResponse(int index) {
+        System.out.println(LINE);
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(list.get(index));
+        System.out.println(LINE);
+    }
+
+    private static void unmarkTask(String input) {
+        int index = Integer.parseInt(input.split("\\s+")[1]) - 1;
+        if (indexNotInList(index)) { return; }
+        list.get(index).unmarkDone();
+        unmarkResponse(index);
+    }
+    private static void unmarkResponse(int index) {
+        System.out.println(LINE);
+        System.out.println("OK! I've marked this task as not done yet:");
+        System.out.println(list.get(index));
+        System.out.println(LINE);
+    }
+    private static void todoTask(String input) {
+        String[] tokens = input.split("\\s+", 2);
+        String task = tokens[1];
+        ToDo t = new ToDo(task);
+        list.add(t);
+        System.out.println(LINE);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(t);
+        System.out.println("Now you have " + list.size() + " tasks in the list");
+        System.out.println(LINE);
+    }
+    private static void deadlineTask(String input) {
+        String[] tokens = input.split("\\s+", 2);
+        String task = tokens[1].split("/")[0];
+        String deadline = tokens[1].split("/")[1];
+        Deadline d = new Deadline(task, deadline);
```

Comment:
Perhaps you might want to avoid using one letter variable names throughout your code? In this case, maybe a variable name like currentDeadline might be better?